relayer/relayer.go

```diff
@@ -160,27 +161,47 @@ func NewRelayer(
 
 func (r *Relayer) processMissedBlocks(
 	sub vms.Subscriber,
+	catchUpBlockHeight *big.Int,
 ) error {
-	// Get the latest processed block height from the database.
+	// Attempt to get the latest processed block height from the database.
+	// Note that the retrieved latest processed block may have already been partially (or fully) processed by the relayer on a previous run. When
+	// processing a warp message in real time, which is when we update the latest processed block in the database, we have no way of knowing
+	// if that is the last warp message in the block
 	latestProcessedBlockData, err := r.db.Get(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey))
 
-	// The following cases are treated as successful:
+	// First, determine the height to process from. There are two cases:
 	// 1) The database contains the latest processed block data for the chain
-	//    - In this case, we parse the block height and process warp logs from that height to the current block
+	//    - In this case, we process from the maximum of the latest processed block and the catch up block height to the latest block
 	// 2) The database has been configured for the chain, but does not contain the latest processed block data
-	//    - In this case, we save the current block height in the database, but do not process any historical warp logs
+	//    - In this case, if a catch up block height is provided, we process from the catch up block height to the latest block
+	//    - Otherwise, we save the current block height in the database, but do not process any historical warp logs
+	var height *big.Int = nil
 	if err == nil {
-		// If the database contains the latest processed block data, then back-process all warp messages from that block to the latest block
-		// Note that the retrieved latest processed block may have already been partially (or fully) processed by the relayer on a previous run. When
-		// processing a warp message in real time, which is when we update the latest processed block in the database, we have no way of knowing
-		// if that is the last warp message in the block
+		// Use the max of the latest processed block and the catch up block height
 		latestProcessedBlock, success := new(big.Int).SetString(string(latestProcessedBlockData), 10)
 		if !success {
 			r.logger.Error("failed to convert latest block to big.Int", zap.Error(err))
 			return err
 		}
+		height = utils.MaxBigInt(latestProcessedBlock, catchUpBlockHeight)
+	} else if errors.Is(err, database.ErrChainNotFound) || errors.Is(err, database.ErrKeyNotFound) {
+		// If the database does not contain the latest processed block data, then we check if a catch up block height is provided.
+		if catchUpBlockHeight != nil {
```

Comment:
It looks like `catchUpBlockHeight` will never be `nil` since we set it using `big.NewInt(0).SetUint64(sourceSubnetInfo.CatchUpBlockHeight)`. Is there a way to distinguish a block height of 0 versus the value being unset?